File: Data Structures & Algorithms/binary-search/submission-2.java
class Solution {
    public int search(int[] nums, int target) {
        int L = 0;
        int R = nums.length-1;

        int mid = L + ((R-L)/2);

        while(L < mid && mid < R){
            
            if(nums[mid] > target){
                R = mid;
            }else if(nums[mid] < target){
                L = mid;
            }else{
                return mid;
            }
            mid = L + ((R-L)/2);
        }

        if(nums[L] == target){
            return L;
        }else if(nums[R] == target){
            return R;
        }
        return -1;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-2.go

func twoSum(nums []int, target int) []int {
    m := make(map[int]int)
    
    for i, v := range nums{
        if h, ok := m[v]; ok {
            return []int{h, i}
        }
        m[target-v] = i
    }
    return []int{}
}
